### ISSUE TITLE
False positive no-member when calling class method on Annotated type

### ISSUE THREAD

**Issue by couteau** [pylint-dev/pylint#9151]
### Bug description

```python
When calling a class method on a typevar defined using typing.Annotated, pylint erroneously reports no-member. 

The following code will reproduce the error:


from typing import Annotated

class Foo:
    @classmethod
    def foo(cls):
        print("foo called")

Bar = Annotated[Foo, int]

Bar.foo()
```
```


### Configuration

_No response_

### Command used

```shell
pylint test.py
```

### Pylint output

```python
************* Module test
test.py:10:0: E1101: Class 'Annotated' has no 'foo' member (no-member)
```

### Expected behavior

Pylint should understand the annotated typevar can be treated the same as the type being annotated and is not of type Annotated

### Pylint version

```shell
pylint 3.0.1
astroid 3.0.0
Python 3.12.0 | packaged by conda-forge | (main, Oct  3 2023, 08:36:57) [Clang 15.0.7 ]
```

### OS / Environment

MacOS Ventura 13.5.2
conda 23.7.4

**Comment by jamescw19** [comment]
I've just encountered the same bug, `pylint==3.0.2`. Looks like it extends further than classmethods to all members of the class:

```python
from typing import Annotated


class A:
    x = "test"

    @classmethod
    def foo(cls):
        pass

    def bar(self):
        pass


AnnotatedA = Annotated[A, "This is a test"]

a = AnnotatedA()
# All lines below raise an error
a.foo()
a.bar()
print(a.x)
```
